include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -185,6 +185,26 @@ transform(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3
         [__op](auto x, auto y, auto& z) { z = __op(x, y); });
 }
 
+// [alg.remove]
+
+template <typename _ExecutionPolicy, typename _Range, typename _UnaryPredicate>
+oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, oneapi::dpl::__internal::__difference_t<_Range>>
+remove_if(_ExecutionPolicy&& __exec, _Range&& __rng, _UnaryPredicate __pred)
+{
+    return oneapi::dpl::__internal::__ranges::__pattern_remove_if(::std::forward<_ExecutionPolicy>(__exec),
+                                                                  views::all(::std::forward<_Range>(__rng)), __pred);
+}
+
+template <typename _ExecutionPolicy, typename _Range, typename _Tp>
+oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, oneapi::dpl::__internal::__difference_t<_Range>>
+remove(_ExecutionPolicy&& __exec, _Range&& __rng, const _Tp& __value)
+{
+    return remove_if(
```

Comment:
I think we can call `oneapi::dpl::__internal::__ranges::__pattern_remove_if` from lower level to not increase dependencies on the same level

Reply:
The current approach is used throughout `glue_algorithm_ranges_impl.h` and `glue_algorithm_impl.h` files. I suggest leaving it for consistency. Otherwise we should refactor many other functions.

Reply:
I see